### PR TITLE
/webapp/boards: fix unit tests

### PR DESCRIPTION
**Pull request by calebroseland**
#### Summary
The `focalboard/mattermost-plugin/webapp`-`focalboard/webapp` merge and subsequent migration to monorepo has caused some issues in execution of `webapp/boards` unit tests. This remedies that by upgrading to Jest 29 and related packages, and ironing out any breaking changes and other assorted issues in moving to monorepo. The fixing-without-upgrading path was explored and did not prove to be worth it.

Very limited changes were made to runtime source code to fix various unit tests and typings:
1. `kanban.tsx`, `kanbanColumnHeader.tsx`, etc. now use `useIntl` instead of `injectIntl` & `intl` prop-passing
2. `editable.tsx` now uses `KeyboardEvent.key` instead of the deprecated `KeyboardEvent.keyCode`
3. `StylesConfig` typing was corrected for a few react-select instances
4. Env typing: upgraded to `react@17`, `react-dom@17`; downgraded to `@types/node@16`.

The remaining changes are all to test files, configs, and dev-deps.

[`jest-fail-on-console`](https://www.npmjs.com/package/jest-fail-on-console) was added to help surface silent/hidden errors—`undefined` Ids , `key`s missing in loops, async/cleanup errors, etc. `react-select-event`, a react-select `userEvent` equivalent is added but presently unused.

6 tests are skipped due to complexity in resolving environmental and breaking changes. Will be addressed after this PR.

Jest configuration moves out of package.json to its own `jest.config.js`.

Jest scripts add an explicit timezone (`Etc/UTC`) for time consistency; `--forceExit --detectOpenHandles --verbose` are  moved to the new `npm run test:debug` so main scripts can support multi-workers.

In the future we'll consider adding [`eslint-plugin-testing-library`](https://github.com/testing-library/eslint-plugin-testing-library) which will assist in modernizing and optimizing testing-library usage. Reports some 700 warnings currently, which was too much for this PR.

#### Ticket Link
None

#### Release Note

```release-note
NONE
```

### CODE REVIEW

**Review by hmhealey — APPROVED** [review]
Looks good to me. Ideally, we wouldn't need to keep the versions of things like React in sync, but it definitely doesn't hurt to do that now

File: webapp/boards/src/components/tutorial_tour_tip/hooks.ts [hunk 1]
```diff
@@ -17,14 +17,14 @@ type PunchoutOffset = {
 export function useMeasurePunchouts(elementIds: string[], additionalDeps: any[], offset?: PunchoutOffset): TutorialTourTipPunchout | null | undefined {
     const elementsAvailable = useElementAvailable(elementIds)
     const [size, setSize] = useState<DOMRect>()
+    // TODO fix throttle memoization
```
[the remark this answers is not in the document]
Comment (calebroseland) [reply webapp/boards/src/components/tutorial_tour_tip/hooks.ts:20]: 100%; glossed over that run-once detail. Thanks for noticing!

File: webapp/boards/tsconfig.json [hunk 2]
```diff
@@ -24,6 +24,8 @@
         "noEmit": true,
         "typeRoots": [ "./src/types", "./node_modules/@types"],
         "paths": {
+            "src": ["src"],
```
[the remark this answers is not in the document]
Comment (calebroseland) [reply webapp/boards/tsconfig.json:27]: Nope, not needed anymore, probably a left-over from playing whack-a-mole.

**Comment by calebroseland** [comment]
Closing this in favor of #22629

**Pull request closed without merging**